### PR TITLE
Correct several issues with xml wallpaper parsing in regolith-set-swaybg script

### PR DESCRIPTION
**Pull request by haaspors**

### CODE REVIEW

**Comment by kgilmer** [comment]
Hi @haaspors , can you briefly describe what this change fixes and how you tested?  @SoumyaRanjanPatnaik FYI.  I don't see any concerns but also don't have much familiarity w/ this aspect of the Wayland integration.

**Comment by haaspors** [comment]
> Hi @haaspors , can you briefly describe what this change fixes and how you tested? @SoumyaRanjanPatnaik FYI. I don't see any concerns but also don't have much familiarity w/ this aspect of the Wayland integration.

1. There was a a CRLF on line 33 which confuses the shell
2. I'm no shell/bash expert, but `local` is only allowed to bash functions I believe.
3. `read -r output_conf` but variable used is `config`
4. Typo `heght:`
5. xpath totally wrong

This makes me think non of this has been run by the original author. So many errors.

I'm using gnome xml background specifications. So, to test, I have this in my `~/.config/regolith3/Xresources`
```
regolith.wallpaper.file: /usr/share/desktop-base/moonlight-theme/wallpaper/gnome-background.xml
```
To test, just use this wallpaper file and run the script, you'll see the script fail.
Here is ouput from whats currently in `3.2-beta2`
```
» /usr/bin/regolith-set-swaybg
/usr/share/desktop-base/moonlight-theme/wallpaper/gnome-background.xml IS valid XML? true
: No such file or directoryg: line 36: /dev/fd/63
```
This first error is because of 1. in my list above.

BTW, here is the xml:
```
<background>
  <static>
    <duration>8640000.0</duration>
    <file>
      <size width="1280" height="720">/usr/share/desktop-base/moonlight-theme/wallpaper/contents/images/1280x720.svg</size>
      <size width="1280" height="1024">/usr/share/desktop-base/moonlight-theme/wallpaper/contents/images/1280x1024.svg</size>
      <size width="1920" height="1080">/usr/share/desktop-base/moonlight-theme/wallpaper/contents/images/1920x1080.svg</size>
      <size width="1920" height="1200">/usr/share/desktop-base/moonlight-theme/wallpaper/contents/images/1920x1200.svg</size>
      <size width="2560" height="1080">/usr/share/desktop-base/moonlight-theme/wallpaper/contents/images/2560x1080.svg</size>
      <size width="2560" height="1440">/usr/share/desktop-base/moonlight-theme/wallpaper/contents/images/2560x1440.svg</size>
      <size width="3840" height="2160">/usr/share/desktop-base/moonlight-theme/wallpaper/contents/images/3840x2160.svg</size>
    </file>
  </static>
</background>
```

File: scripts/regolith-set-swaybg [hunk 1]
```diff
@@ -17,21 +17,18 @@ fi
 echo "$wallpaper IS valid XML? $is_valid_xml"
 
 if [[ "$wallpaper" == "" || ! $is_valid_xml ]]; then
-	exec swaymsg output "*" background "$color" solid_color
+  exec swaymsg output "*" background "$color" solid_color
 fi
 
-if $is_valid_xml; then
+if [[ $is_valid_xml ]]; then
```
Comment (SoumyaRanjanPatnaik) [inline scripts/regolith-set-swaybg:23]: This doesn't work. The control flow goes inside the XML branch, with this change.

**Review by SoumyaRanjanPatnaik — CHANGES_REQUESTED** [review]

**Comment by SoumyaRanjanPatnaik** [comment]
@haaspors can you test this PR with non XML wallpapers as well?

File: scripts/regolith-set-swaybg (hunk 1 above)
Comment (haaspors) [reply scripts/regolith-set-swaybg]: Right, my bash is a bit rusty, thanks.

**Comment by kgilmer** [comment]
I've added some minor tweaks on top of @haaspors 's fixes in https://github.com/regolith-linux/regolith-wm-config/pull/45.  I've tested both xml and image backgrounds.

**Comment by kgilmer** [comment]
Thanks for the fixes @haaspors !

**Pull request merged**